system-addon/lib/FaviconFeed.jsm

```diff
@@ -50,15 +54,20 @@ this.FaviconFeed = class FaviconFeed {
     let {data, etag, status} = await this.loadFromURL(this.prefs.get("tippyTop.service.endpoint"), headers);
     if (status === 304) {
       // Not modified, we are done. No need to dispatch actions or update cache. We'll check again in 24+ hours.
-      this.tippyTopLastUpdated = Date.now();
-      return;
-    }
-    if (status === 200 && data) {
+      this.tippyTopNextUpdate = Date.now() + TIPPYTOP_UPDATE_TIME;
+      this.numRetries = 0;
```

Comment:
This got me wondering if we should be storing some of this state in the cache? In particular, in the common case of `304`, `_timestamp` isn't updated, so every restart (after 24 hours from first `200`) will trigger a refresh?

Reply:
yeah, good point. We should probably store the next update time at least on 304 and 200. Not sure if it's worth writing to cache on every failed retry though?

Reply:
Ehh.. mm.. I suppose just updating `_timestamp` on `304` should be enough and not bother caching with `numRetries`, i.e., a restart resets the backoff to 5 minutes.